rust-runtime/aws-smithy-runtime-api/src/client/http/request.rs

```diff
@@ -404,8 +404,10 @@ impl Headers {
     /// Removes all headers with a given key
     ///
     /// If there are multiple entries for this key, the first entry is returned
```

Comment:
Should we be more explicit and add something like "all other entries are discarded" or "to get all values for a given key, keep calling this until it returns None"?